dash/include/dash/matrix/internal/Matrix-inl.h

```diff
@@ -478,7 +479,8 @@ constexpr typename Matrix<T, NumDim, IndexT, PatternT, LocalMemT>::const_iterato
 Matrix<T, NumDim, IndexT, PatternT, LocalMemT>
 ::end() const noexcept
 {
-  return const_iterator(_begin + _size);
+    return const_iterator(
+        const_cast<GlobMem_t *>(&_glob_mem), _pattern, _begin.pos() + _size);
```

Comment:
Why not use the copy-constructor here? (and above)

Reply:
Because we cannot assign a `const_iterator` to an `iterator`. They are two different types and also the template parameters are different. Actually, const correctness is still a burden because we do not really implement it the right way.